### ISSUE TITLE
Add native-image support for TransactionalEventListener

### ISSUE THREAD

**Issue by mhalbritter** [spring-projects/spring-framework#28993]
When running the [transactional-event-listener sample](https://github.com/spring-projects/spring-aot-smoke-tests/tree/main/transactional-event-listener), it fails in a native-image with:

```
org.springframework.beans.factory.BeanInitializationException: Failed to process @EventListener annotation on bean with name 'transactionalEventListener': Proxy class defined by interfaces [interface org.springframework.transaction.event.TransactionalEventListener, interface org.springframework.core.annotation.SynthesizedAnnotation] not found. Generating proxy classes at runtime is not supported. Proxy classes need to be defined at image build time by specifying the list of interfaces that they implement. To define proxy classes use -H:DynamicProxyConfigurationFiles=<comma-separated-config-files> and -H:DynamicProxyConfigurationResources=<comma-separated-config-resources> options.
	at org.springframework.context.event.EventListenerMethodProcessor.afterSingletonsInstantiated(EventListenerMethodProcessor.java:158) ~[transactional-event-listener:na]
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.preInstantiateSingletons(DefaultListableBeanFactory.java:942) ~[transactional-event-listener:6.0.0-SNAPSHOT]
	at org.springframework.context.support.AbstractApplicationContext.finishBeanFactoryInitialization(AbstractApplicationContext.java:926) ~[transactional-event-listener:6.0.0-SNAPSHOT]
	at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:592) ~[transactional-event-listener:6.0.0-SNAPSHOT]
	at org.springframework.boot.SpringApplication.refresh(SpringApplication.java:731) ~[transactional-event-listener:3.0.0-SNAPSHOT]
	at org.springframework.boot.SpringApplication.refreshContext(SpringApplication.java:430) ~[transactional-event-listener:3.0.0-SNAPSHOT]
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:312) ~[transactional-event-listener:3.0.0-SNAPSHOT]
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1303) ~[transactional-event-listener:3.0.0-SNAPSHOT]
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1292) ~[transactional-event-listener:3.0.0-SNAPSHOT]
	at com.example.transactional.eventlistener.TransactionalEventListenerApplication.main(TransactionalEventListenerApplication.java:10) ~[transactional-event-listener:na]
Caused by: com.oracle.svm.core.jdk.UnsupportedFeatureError: Proxy class defined by interfaces [interface org.springframework.transaction.event.TransactionalEventListener, interface org.springframework.core.annotation.SynthesizedAnnotation] not found. Generating proxy classes at runtime is not supported. Proxy classes need to be defined at image build time by specifying the list of interfaces that they implement. To define proxy classes use -H:DynamicProxyConfigurationFiles=<comma-separated-config-files> and -H:DynamicProxyConfigurationResources=<comma-separated-config-resources> options.
	at com.oracle.svm.core.util.VMError.unsupportedFeature(VMError.java:89) ~[na:na]
	at com.oracle.svm.reflect.proxy.DynamicProxySupport.getProxyClass(DynamicProxySupport.java:158) ~[na:na]
	at java.lang.reflect.Proxy.getProxyConstructor(Proxy.java:48) ~[transactional-event-listener:na]
	at java.lang.reflect.Proxy.newProxyInstance(Proxy.java:1037) ~[transactional-event-listener:na]
	at org.springframework.core.annotation.SynthesizedMergedAnnotationInvocationHandler.createProxy(SynthesizedMergedAnnotationInvocationHandler.java:305) ~[na:na]
	at org.springframework.core.annotation.TypeMappedAnnotation.createSynthesizedAnnotation(TypeMappedAnnotation.java:344) ~[na:na]
	at org.springframework.core.annotation.AbstractMergedAnnotation.synthesize(AbstractMergedAnnotation.java:210) ~[transactional-event-listener:6.0.0-SNAPSHOT]
	at org.springframework.core.annotation.AbstractMergedAnnotation.synthesize(AbstractMergedAnnotation.java:200) ~[transactional-event-listener:6.0.0-SNAPSHOT]
	at org.springframework.core.annotation.AnnotatedElementUtils.findMergedAnnotation(AnnotatedElementUtils.java:634) ~[na:na]
	at org.springframework.transaction.event.TransactionalApplicationListenerMethodAdapter.<init>(TransactionalApplicationListenerMethodAdapter.java:67) ~[na:na]
	at org.springframework.transaction.event.TransactionalEventListenerFactory.createApplicationListener(TransactionalEventListenerFactory.java:56) ~[transactional-event-listener:6.0.0-SNAPSHOT]
	at org.springframework.context.event.EventListenerMethodProcessor.processBean(EventListenerMethodProcessor.java:200) ~[transactional-event-listener:na]
	at org.springframework.context.event.EventListenerMethodProcessor.afterSingletonsInstantiated(EventListenerMethodProcessor.java:154) ~[transactional-event-listener:na]
	... 9 common frames omitted
```

**Comment by sdeleuze** [comment]
Should be fixed by #29041 if we decide to do it.

**Comment by sbrannen** [comment]
Superseded by #29041.

Synthesized annotation proxies no longer implement `SynthesizedAnnotation`.

Please let us know if you still run into any issues with the latest Spring Framework snapshot builds.
